### DISCUSSION TITLE
The `peer` class does not exist error

### DISCUSSION THREAD

**Discussion by yabdab** [tailwindlabs/tailwindcss#5032] [Help]
Trying to use peer ( _in an @apply declaration_ ) in my scss file , but when building postcss throws this error...

```
The `peer` class does not exist. If `peer` is a custom class, make sure it is defined within a `@layer` directive.
```

I am using  ...

postcss@8.3.5
 tailwindcss@2.2.4

Here is the @apply

```css
@apply peer h-10 w-full border-b-2 border-gray-300 text-gray-900 placeholder-transparent focus:outline-none focus:border-rose-600;
```

What am I doing wrong?

**Comment by yabdab** [comment]
Furthermore, it looks like none of the new  v 2.2.x classes are working with @apply?

**Reply by yabdab** [reply]
example: https://play.tailwindcss.com/Y8J0ZzvRGT

**Comment by simonswiss** [comment]
Hey!

It's because the `peer` class actually _does not_ exist. It's being used a _construct_ and referred in the `peer-*:` variant classes, but there is no CSS output for the `peer` class itself.

Think of it more as a "marker" or "ref" in your HTML that peer variants can listen to.

The exact same happens with the `group` class. It's not a class by itself, but is more use like a reference point for `group-*` variants to be used.

This is why you're getting the "The `peer` class does not exist" error message. It indeed does not exist! Hope my explanation makes sense.

You can still use the `peer-*` variants in `@apply`, **but your `peer` class must be applied in the HTML**. 

Here's an example of that:

https://user-images.githubusercontent.com/485747/126598789-7558140a-b669-4bd7-8cbd-408601181814.mov

https://play.tailwindcss.com/0k1w5yqBhJ?file=css

Try hovering over the first square (which has the `peer` class on the HTML). The second square, which has the `peer-hover:` variant, will change color!

The example above also uses a new class form Tailwind 2.2 in the `@apply` rule: the color-opacity shorthand syntax. As you can see, it works.

**Reply by yabdab** [reply]
Simon,

Thanks! Your explanation makes total sense. I now know how to handle this going forward. 😁

**Reply by yabdab** [reply]
It looks like the placeholder peer  classes are not obeyed when used in @apply ...

https://play.tailwindcss.com/DnuE7aaGyd?file=css

Notice how Email Address input stays the same ?
